lib/vector-core/src/mapping/query/function/not.rs

```diff
@@ -16,7 +16,7 @@ impl Function for NotFn {
         self.query.execute(ctx).and_then(|v| match v {
             QueryValue::Value(Value::Boolean(b)) => Ok(Value::Boolean(!b).into()),
             QueryValue::Value(v) => Err(format!("unable to perform NOT on {:?} value", v)),
-            v => Err(format!("unable to perform NOT on {:?} value", v)),
+            v @ QueryValue::Regex(_) => Err(format!("unable to perform NOT on {:?} value", v)),
```

Comment:
Is there a clippy lint against *any* wildcard in matches? That seems excessive.

Reply:
Agreed. I think this is https://rust-lang.github.io/rust-clippy/master/#wildcard_enum_match_arm . In this case, I think `if let Some()` would work.

Reply:
Yeah, `clippy::pedantic` doesn't love wildcards. We can back this off if we like being able to do this pattern, which seems good to me overall.

Reply:
Reversed in d949fa1